ampersand-view.js

```diff
@@ -375,8 +393,46 @@ assign(View.prototype, {
         var collectionView = new CollectionView(config);
         collectionView.render();
         return this.registerSubview(collectionView);
+    },
+
+    _setRender: function(obj) {
+        Object.defineProperty(obj, 'render', {
+            get: function() {
+                return this.__render;
+            },
+            set: function(fn) {
+                if (typeof fn !== 'function') {
+                    throw new ReferenceError('render() must be a function. ' +
+                        'received: ' + fn.toString());
+                }
+                this.__render = function() {
+                    fn.apply(this, arguments);
+                    return this._render();
+                };
+            }
+        });
+    },
+
+    _setRemove: function(obj) {
+        Object.defineProperty(obj, 'remove', {
+            get: function() {
+                return this.__remove;
+            },
+            set: function(fn) {
+                if (typeof fn !== 'function') {
```

Comment:
True, but we don't really do this kind of thing elsewhere though, right? We don't check if `remove` if function, for example. If we were to do this here, we should do it elsewhere. I don't think this case is different enough to warrant this check, personally.